operators/src/processing/raster_vector_join/points_aggregated.rs

```diff
@@ -0,0 +1,323 @@
+use futures::stream::BoxStream;
+use futures::{StreamExt, TryStreamExt};
+
+use geoengine_datatypes::raster::{GridIndexAccess, Pixel, RasterDataType};
+use geoengine_datatypes::{
+    collections::{
+        FeatureCollectionInfos, FeatureCollectionModifications, GeometryRandomAccess,
+        MultiPointCollection,
+    },
+    raster::Raster,
+};
+
+use crate::engine::{
+    QueryContext, QueryProcessor, QueryRectangle, RasterQueryProcessor, TypedRasterQueryProcessor,
+    VectorQueryProcessor,
+};
+use crate::processing::raster_vector_join::aggregator::{
+    Aggregator, FirstValueDecimalAggregator, FirstValueNumberAggregator, MeanValueAggregator,
+    TypedAggregator,
+};
+use crate::processing::raster_vector_join::util::FeatureTimeSpanIter;
+use crate::processing::raster_vector_join::AggregationMethod;
+use crate::util::Result;
+use geoengine_datatypes::primitives::MultiPointAccess;
+
+pub struct RasterPointAggregateJoinProcessor {
+    points: Box<dyn VectorQueryProcessor<VectorType = MultiPointCollection>>,
+    raster_processors: Vec<TypedRasterQueryProcessor>,
+    column_names: Vec<String>,
+    aggregation: AggregationMethod,
+}
+
+impl RasterPointAggregateJoinProcessor {
+    pub fn new(
+        points: Box<dyn VectorQueryProcessor<VectorType = MultiPointCollection>>,
+        raster_processors: Vec<TypedRasterQueryProcessor>,
+        column_names: Vec<String>,
+        aggregation: AggregationMethod,
+    ) -> Self {
+        Self {
+            points,
+            raster_processors,
+            column_names,
+            aggregation,
+        }
+    }
+
+    async fn extract_raster_values<P: Pixel>(
+        points: &MultiPointCollection,
+        raster_processor: &dyn RasterQueryProcessor<RasterType = P>,
+        new_column_name: &str,
+        aggregation: AggregationMethod,
+        query: QueryRectangle,
+        ctx: &dyn QueryContext,
+    ) -> Result<MultiPointCollection> {
+        let mut aggregator = Self::create_aggregator::<P>(points.len(), aggregation);
+
+        let points = points.sort_by_time_asc()?;
+
+        for time_span in FeatureTimeSpanIter::new(points.time_intervals()) {
+            let query = QueryRectangle {
+                bbox: query.bbox,
+                time_interval: time_span.time_interval,
+                spatial_resolution: query.spatial_resolution,
+            };
+
+            let mut rasters = raster_processor.raster_query(query, ctx)?;
+
+            // TODO: optimize geo access (only specific tiles, etc.)
```

Comment:
can you estimate thhe impact of this?

Reply:
I don't know, I guess I already covered some low hanging fruits.
The rest is very dependent on the concrete case.
I guess we could implement some heuristics, but we first have to find some scenarios. We would need to benchmark them prior and then optimize.